web/src/main/java/com/marklogic/hub/web/service/NewJobService.java

```diff
@@ -228,16 +228,16 @@ private void createJobs(List<JobModel> jobModels, JSONObject jobJson) {
             Optional.ofNullable(step.getOptions()).ifPresent(o -> js.targetDatabase = ((TextNode) o.getOrDefault("targetDatabase", new TextNode(""))).asText());
 
             JsonNode res = stepRes.get(key);
-            js.totalEvents = res.get("totalEvents").asLong();
-            js.successfulEvents = res.get("successfulEvents").asLong();
-            js.failedEvents = res.get("failedEvents").asLong();
+            js.totalEvents = res.get("totalEvents") != null ? res.get("totalEvents").asLong() : 0;
```

Comment:
I recommend using:

    res.has("totalEvents")

The check for a long value is repeated enough times that a separate method to handle it would be useful - e.g. private long getLong(JsonNode node, String propertyName).

Reply:
OK, agree.  @rjrudin👍